### ISSUE TITLE
Newest Update 1.2.0 Game Breaking

### ISSUE THREAD

**Issue by TheMasterKy** [BeeIsYou/PeristentPurchases#3]
When the game resets by either not meeting quota or using /restart (Quick Restart Mod), it will not move past the (You failed to meet quota) screen. My only fix was reverting the update back to 1.1.0

Just wanted to make you aware

**Comment by Tae47** [comment]
I have the same problem, seems to not be a compatibility issue, but just the mod itself.

**Comment by TheMasterKy** [comment]
> I have the same problem, seems to not be a compatibility issue, but just the mod itself.

I have reverted back to the version before it and it works perfectly fine. Not sure what their update was supposed to fix but it did the opposite of fix anything

**Comment by MuffinCrow** [comment]
Yeah also found the issue. To add, once I restarted the game, it kept our walkies, flashlights, radar-booster, keys, spray paint, and shotgun shells but did not keep the plushie that we had so idk what happened but it doesn't really work rn

**Comment by Gflo** [comment]
``Error  : Unity Log] ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
Stack trace:
System.Collections.Generic.List`1[T].get_Item (System.Int32 index) (at <787acc3c9a4c471ba7d971300105af24>:0)
PersistentPurchases.Patches.storeUnlocked (StartOfRound __instance, System.Collections.Generic.List`1[System.Tuple`3[System.Int32,UnityEngine.Vector3,UnityEngine.Vector3]]& __state) (at <9d36770362664d9fa32dc014c89006df>:0)
(wrapper dynamic-method) StartOfRound.DMD<StartOfRound::ResetShip>(StartOfRound)
StartOfRound+<playersFiredGameOver>d__263.MoveNext () (at <44743d9474784365a095189c76175301>:0)
UnityEngine.SetupCoroutine.InvokeMoveNext (System.Collections.IEnumerator enumerator, System.IntPtr returnValueAddress) (at <e27997765c1848b09d8073e5d642717a>:0)``


Getting this at the fired screen and stuck there as well, v1.2.0

**Comment by TheMasterKy** [comment]
Really sucks to see there is no attention from the dev on this issue or is really a better way to report or contact.

**Comment by Venaf** [comment]
Very similar error here, but slightly different details, if it helps:

[Info   :PersistentPurchases] Taking note of bought unlockables
[Error  : Unity Log] ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
Stack trace:
System.Collections.Generic.List`1[T].get_Item (System.Int32 index) (at <787acc3c9a4c471ba7d971300105af24>:IL_0009)
PersistentPurchases.Patches.storeUnlocked (StartOfRound __instance, System.Collections.Generic.List`1[System.Tuple`3[System.Int32,UnityEngine.Vector3,UnityEngine.Vector3]]& __state) (at <9d36770362664d9fa32dc014c89006df>:IL_0049)
(wrapper dynamic-method) StartOfRound.DMD<StartOfRound::ResetShip>(StartOfRound)
StartOfRound+<playersFiredGameOver>d__263.MoveNext () (at <44743d9474784365a095189c76175301>:IL_02D5)
UnityEngine.SetupCoroutine.InvokeMoveNext (System.Collections.IEnumerator enumerator, System.IntPtr returnValueAddress) (at <e27997765c1848b09d8073e5d642717a>:IL_0026)

**Comment by TheMasterKy** [comment]
I just noticed someone made a reverted copy of the working version of this mod. To whoever did that. Thank you! For anyone still wanting to use the mod without it telling you to update every time you load it up if using a mod manager. Search for Persistent Purchases and download the other one that isn't made by TheBeeTeam

**Comment by CherenTheCat** [comment]
> I just noticed someone made a reverted copy of the working version of this mod. To whoever did that. Thank you! For anyone still wanting to use the mod without it telling you to update every time you load it up if using a mod manager. Search for Persistent Purchases and download the other one that isn't made by TheBeeTeam

Seems that the backport stopped working with v47, so the team that posted the backport took it down. Hopefully someone can make a fixed version, this mod is probably the one I most want to get updated.

**Comment by TheMasterKy** [comment]
I 100% agree. It's all broken. I've reached out to the owner of the mod but they are not responding to me relating to the mod at all. I'll continue making attempts to get this worked out and hopefully it will

-------- Original Message --------
